src/Workspaces/CSharp/Portable/Formatting/Rules/SuppressFormattingRule.cs

```diff
@@ -286,6 +289,109 @@ private void AddStatementExceptBlockSuppressOperations(List<SuppressOperation> l
+                {
+                    return;
+                }
+
+                var startToken = pragmaWarningDirectiveTrivia.GetLastToken(includeZeroWidth: true, includeDirectives: true);
+
+                var endDirective = pragmaWarningDirectiveTrivia.GetNextDirective(trivia => IsFormatDirective(trivia, SyntaxKind.RestoreKeyword));
+                var endToken = endDirective is null
+                    ? structure.SyntaxTree.GetRoot().GetLastToken(includeZeroWidth: true)
+                    : endDirective.GetFirstToken(includeDirectives: true);
+
+                if (startToken.IsKind(SyntaxKind.None) || endToken.IsKind(SyntaxKind.None))
+                {
+                    return;
+                }
+
+                var startTokenForWrappingSuppression = pragmaWarningDirectiveTrivia.GetFirstToken().GetPreviousToken();
+                var startSpanForWrappingSuppression = startTokenForWrappingSuppression.IsKind(SyntaxKind.None)
+                    ? pragmaWarningDirectiveTrivia.FullSpan.Start
+                    : startTokenForWrappingSuppression.Span.End;
+                var endTokenForWrappingSuppression = endToken.GetNextToken();
+                var endSpanForWrappingSuppression = endTokenForWrappingSuppression.IsKind(SyntaxKind.None)
+                    ? endToken.FullSpan.End
+                    : endTokenForWrappingSuppression.SpanStart;
+
+                // Spacing is suppressed within the #pragma region. Wrapping is suppressed from the non-trivia token
+                // preceding the #pragma region to the non-trivia token following the #pragma region.
+                var textSpan = TextSpan.FromBounds(startToken.Span.End, endToken.SpanStart);
+                var textSpanForWrappingSuppression = TextSpan.FromBounds(startSpanForWrappingSuppression, endSpanForWrappingSuppression);
+                list.Add(new SuppressOperation(startToken, endToken, textSpan, SuppressOption.NoSpacing));
+                list.Add(new SuppressOperation(startToken, endToken, textSpanForWrappingSuppression, SuppressOption.NoWrapping));
+            }
+        }
+
+        private static bool IsFormatDirective(DirectiveTriviaSyntax trivia, SyntaxKind disableOrRestoreKeyword)
+        {
+            if (!(trivia is PragmaWarningDirectiveTriviaSyntax pragmaWarningDirectiveTrivia))
+            {
+                return false;
+            }
+
+            if (!pragmaWarningDirectiveTrivia.DisableOrRestoreKeyword.IsKind(disableOrRestoreKeyword))
+            {
+                return false;
+            }
+
+            foreach (var errorCode in pragmaWarningDirectiveTrivia.ErrorCodes)
+            {
+                if (!(errorCode is IdentifierNameSyntax identifierName))
+                {
+                    continue;
+                }
+
+                if (identifierName.Identifier.ValueText.Equals("format")
+                    || identifierName.Identifier.ValueText.Equals("IDE0055"))
```

Comment:
duplicated constants.  consider moving to central location and referencing appropriately.

Reply:
Done!